mozci/push.py

```diff
@@ -725,8 +725,9 @@ def _iterate_failures(self, runnable_type, max_depth=None):
                 ) and not any(result is True for result in classified_as_cause[name]):
                     del adjusted_candidate_regressions[name]
 
-                elif name in passing_runnables and not any(
-                    result is True for result in classified_as_cause[name]
+                elif name in passing_runnables and (
+                    not any(result is True for result in classified_as_cause[name])
+                    or any(result is False for result in classified_as_cause[name])
```

Comment:
The first clause of that `or` statement implies the second. So you can get away with just the second.

Reply:
There are a few cases where the first doesn't imply the second.
E.g. when `classified_as_cause[name]` contains both True and False (the first is false, the second is true), or when it contains only False (the first is true, the second is false)

It'd probably be clearer if we used an enum instead of a bool to specify whether the push was classified as cause or if another one was...